tools/cachemgr.cc

```diff
@@ -861,6 +861,11 @@ process_request(cachemgr_request * req)
 
     Ip::Address::FreeAddr(AI);
 
+    // XXX: This Squid does not support receiving cache_object requests, but
+    // very old Squid versions do not support cache manager requests with an
+    // http scheme. We are using cache_object here per "be very conservative"
+    // recommendation archived at
+    // http://lists.squid-cache.org/pipermail/squid-dev/2023-January/009848.html
     l = snprintf(buf, sizeof(buf),
                  "GET cache_object://%s/%s%s%s HTTP/1.0\r\n"
```

Comment:
@yadij, please check whether this code and its justification matches your _actual_ [recommendation](http://lists.squid-cache.org/pipermail/squid-dev/2023-January/009848.html). If they do not match, please suggest what should be done with this code instead. Please be as specific as possible. Thank you.

FWIW, I recommend upgrading this code to send `http`-based URLs, but the current plan is to follow your recommendation instead.

Reply:
> Second best plan is to simply convert cachemgr.cgi to http://-only and mark the complex change with XXX for volunteer improvement later (if wanted). This is necessary to remove a lot of the pain caused by lack of (a), and relies on (b) for further reductions

@eduard-bagdasaryan, I recommend using the above "second best" plan for now. If somebody decides to add `http` scheme support detection, they would be in a much better position to test that code against the old Squids they care about.

Reply:
> Second best plan is to simply convert cachemgr.cgi to http://-only and mark the complex change with XXX

Ok, let's make this change - it looks like there is a consensus about that.

Reply:
The split of src/ changes and tools/ changes into separate PRs still remains to complete this item.

>  b) leaving the squid binary removal a Squid version cycle. Giving admin some time to upgrade their tooling.